unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/Passport/Passport/PlayerPassportHUDController.cs

```diff
@@ -0,0 +1,90 @@
+using Cysharp.Threading.Tasks;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using UnityEngine;
+using DCL;
+using DCL.Helpers;
+using SocialFeaturesAnalytics;
+
+public class PlayerPassportHUDController : IHUD
+{
+    internal readonly PlayerPassportHUDView view;
+    internal readonly StringVariable currentPlayerId;
+    internal readonly IUserProfileBridge userProfileBridge;
+    internal UserProfile currentUserProfile;
+
+    private readonly ISocialAnalytics socialAnalytics;
+
+    private PassportPlayerInfoComponentController playerInfoController;
+    private PassportPlayerPreviewComponentController playerPreviewController;
+    private PassportNavigationComponentController passportNavigationController;
+
+    public PlayerPassportHUDController(
+        PlayerPassportHUDView view,
+        PassportPlayerInfoComponentController playerInfoController,
+        PassportPlayerPreviewComponentController playerPreviewController,
+        PassportNavigationComponentController passportNavigationController,
+        StringVariable currentPlayerId,
+        IUserProfileBridge userProfileBridge,
+        ISocialAnalytics socialAnalytics)
+    {
+        this.view = view;
+        this.playerInfoController = playerInfoController;
+        this.playerPreviewController = playerPreviewController;
+        this.passportNavigationController = passportNavigationController;
+        this.currentPlayerId = currentPlayerId;
+        this.userProfileBridge = userProfileBridge;
+        this.socialAnalytics = socialAnalytics;
+
+        view.Initialize(() => currentPlayerId.Set(null));
```

Comment:
instead of exposing a unityAction in the initialization, create a new event in the view `OnClosed`